pallets/rc-migrator/src/lib.rs

```diff
@@ -0,0 +1,2581 @@
+					let weight = AccountsMigrator::<T>::obtain_rc_accounts();
+					weight_counter.consume(weight);
+					let total_issuance = <T as Config>::Currency::total_issuance();
+					RcMigratedBalance::<T>::mutate(|tracker| {
+						// initialize `kept` balance as total issuance, we'll substract from it as
+						// we migrate accounts
+						tracker.kept = total_issuance;
+						tracker.migrated = 0;
+					});
+					Self::deposit_event(Event::MigratedBalanceRecordSet {
+						kept: total_issuance,
+						migrated: 0,
+					});
+					Self::transition(MigrationStage::AccountsMigrationOngoing { last_key: None });
+				},
+				MigrationStage::AccountsMigrationOngoing { last_key } => {
+					let res =
+						with_transaction_opaque_err::<Option<T::AccountId>, Error<T>, _>(|| {
+							match AccountsMigrator::<T>::migrate_many(last_key, &mut weight_counter)
+							{
+								Ok(last_key) => TransactionOutcome::Commit(Ok(last_key)),
+								Err(e) => TransactionOutcome::Rollback(Err(e)),
+							}
+						})
+						.expect("Always returning Ok; qed");
+
+					match res {
+						Ok(None) => {
+							// accounts migration is completed
+							Self::transition(MigrationStage::AccountsMigrationDone);
+						},
+						Ok(Some(last_key)) => {
+							// accounts migration continues with the next block
+							Self::transition(MigrationStage::AccountsMigrationOngoing {
+								last_key: Some(last_key),
+							});
+						},
+						Err(err) => {
+							defensive!("Error while migrating accounts: {:?}", err);
+							// stage unchanged, retry.
+						},
+					}
+				},
+				MigrationStage::AccountsMigrationDone => {
+					AccountsMigrator::<T>::finish_balances_migration();
+					// Note: swap this out for faster testing to skip some migrations
+					Self::transition(MigrationStage::MultisigMigrationInit);
+				},
+				MigrationStage::MultisigMigrationInit => {
+					Self::transition(MigrationStage::MultisigMigrationOngoing { last_key: None });
+				},
+				MigrationStage::MultisigMigrationOngoing { last_key } => {
+					let res = with_transaction_opaque_err::<Option<_>, Error<T>, _>(|| {
+						match MultisigMigrator::<T>::migrate_many(
+							last_key,
+							&mut weight_counter,
+						) {
+							Ok(last_key) => TransactionOutcome::Commit(Ok(last_key)),
+							Err(e) => TransactionOutcome::Rollback(Err(e)),
+						}
+					})
+					.expect("Always returning Ok; qed");
+
+					match res {
+						Ok(None) => {
+							// multisig migration is completed
+							Self::transition(MigrationStage::MultisigMigrationDone);
+						},
+						Ok(Some(last_key)) => {
+							// multisig migration continues with the next block
+							Self::transition(MigrationStage::MultisigMigrationOngoing {
+								last_key: Some(last_key),
+							});
+						},
+						e => {
+							defensive!("Error while migrating multisigs: {:?}", e);
+						},
+					}
+				},
+				MigrationStage::MultisigMigrationDone => {
+					Self::transition(MigrationStage::ClaimsMigrationInit);
```

Comment:
void step, but yeah I can see that you want to keep the steps per pallet consistent :)

Reply:
Kind of. If we trim it down then it will be like 30 blocks faster, so maybe not worth it for loosing the pattern.

Reply:
I would leave it as-is to not touch sensitive code last minute, just sharing my observation here.